### PR TITLE
handle null dates in document_match, closes#161

### PR DESCRIPTION
**Pull request by orischwartz**

### CODE REVIEW

**Comment by orischwartz** [comment]
No idea why this fails

File: app/views/certifications/_document_match.html.erb [hunk 1]
```diff
@@ -8,6 +8,6 @@
 <tr id="<%= name.downcase.parameterize %>-match">
   <td> <%= match ? svg_icon(:found) : svg_icon(:missing) %> </td>
   <td> <%= name %> </td>
-  <td> <%= date.to_formatted_s(:short_date) %> </td>
+  <td> <%= date ? date.to_formatted_s(:short_date) : "" %> </td>
```
Comment (shanear) [inline app/views/certifications/_document_match.html.erb]: we should deal with this in the code, not the template.

Comment (orischwartz) [inline app/views/certifications/_document_match.html.erb]: @shanear Yeah I agree...this was rushed after being uncovered during testing last week. I thought it was a launch blocker but I now don't believe you can reach this state unless you enter the URL of an appeal that is not ready for certification. So hardly a priority. 

I'm thinking detect if appeal is missing any of `[form9, nod, soc]_date` in [CertificationsController#appeal](https://github.com/department-of-veterans-affairs/caseflow-certification/blob/master/app/controllers/certifications_controller.rb#L73) and alert user. Thoughts?

Comment (shanear) [inline app/views/certifications/_document_match.html.erb]: i talked with Kavi about this and he said it shouldn't happen, but if it does, we should have a generic "Case isn't ready to certify: go check VACOLS" screen.

Sounds great, maybe can add the check in something like `appeal.missing_certification_data?` in `CertificationsController.new` just like we're doing with the other checks.

**Comment by shanear** [comment]
:+1: when travis gives us the :white_check_mark:

**Pull request merged**